src/storage/mvcc/reader/scanner/util.rs

```diff
@@ -81,7 +83,29 @@ where
     assert!(write.short_value.is_none());
     let seek_key = user_key.clone().append_ts(write.start_ts);
     default_cursor.near_seek(&seek_key, &mut statistics.data)?;
-    assert!(default_cursor.valid());
+    if !default_cursor.valid() {
+        CRITICAL_ERROR
+            .with_label_values(&["default value not found"])
+            .inc();
+        if panic_when_unexpected_key_or_data() {
+            set_panic_mark();
+            panic!(
+                "default value not found for key {}, write: {:?}",
+                hex::encode_upper(&user_key.to_raw()?),
```

Comment:
How about use escape?

Reply:
Shouldn't we use new hex format for key?